_1327/documents/views.py

```diff
@@ -521,12 +538,16 @@ def delete_autosave(request, title):
 
 	# first check that the user actually may change this document
 	document = get_object_or_404(Document, url_title=title)
-	check_permissions(document, request.user, [document.edit_permission_name])
 
 	# second check that the supplied autosave id matches to the document and has been created by the user
 	autosave_id = request.POST['autosave_id']
 	autosave = get_object_or_404(TemporaryDocumentText, id=autosave_id)
 	autosaves_for_object_and_user = TemporaryDocumentText.objects.filter(document=document, author=request.user)
+
+	# a new document does not have permissions, just check if the autosave author is correct
```

Comment:
this comment does not explain why a permission check is done nevertheless if the author does not match